### DISCUSSION TITLE
vscodehtml: Failed to find project root, language server won't start.

### DISCUSSION THREAD

**Discussion by gnietto** [w0rp/ale#4608] [Q&A - Ask for help with problems] [answered]
I feel I'm missing something very basic here.
I'm using https://github.com/hrsh7th/vscode-langservers-extracted LSP servers.
The hrsh7th's CSS LSP server is working okay, but HTML LSP server shows the error message: Failed to find project root, language server won't start.

My :ALEInfo is: 

    Current Filetype: html
    Available Linters: ['alex', 'angular', 'cspell', 'eslint', 'fecs', 'htmlhint', 'proselint', 'stylelint', 'tidy', 'vscodehtml', 'writegood']
       Linter Aliases:
    'angular' -> ['angular-language-server']
    'writegood' -> ['write-good']
      Enabled Linters: ['vscodehtml']
      Ignored Linters: []
     Suggested Fixers: 
      'fecs' - Apply fecs format to a file.
      'html-beautify' - Fix HTML files with html-beautify from js-beautify.
      'prettier' - Apply prettier to a file.
      'remove_trailing_lines' - Remove all blank lines at the end of a file.
      'rustywind' - Sort Tailwind CSS classes
      'tidy' - Fix HTML files with tidy.
      'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
     Linter Variables:
    
     Global Variables:
    
    let g:ale_cache_executable_check_failures = v:null
    let g:ale_change_sign_column_color = 0
    let g:ale_command_wrapper = v:null
    let g:ale_completion_delay = 100
    let g:ale_completion_enabled = 1
    let g:ale_completion_max_suggestions = 50
    let g:ale_disable_lsp = 0
    let g:ale_echo_cursor = 1
    let g:ale_echo_msg_error_str = 'Error'
    let g:ale_echo_msg_format = '%code: %%s'
    let g:ale_echo_msg_info_str = 'Info'
    let g:ale_echo_msg_warning_str = 'Warning'
    let g:ale_enabled = 1
    let g:ale_fix_on_save = 0
    let g:ale_fixers = {'json': ['prettier'], 'html': ['prettier'], 'javascript': ['prettier'], 'css': ['prettier']}
    let g:ale_history_enabled = 1
    let g:ale_history_log_output = 1
    let g:ale_keep_list_window_open = 0
    let g:ale_lint_delay = 200
    let g:ale_lint_on_enter = 1
    let g:ale_lint_on_filetype_changed = 1
    let g:ale_lint_on_insert_leave = 1
    let g:ale_lint_on_save = 1
    let g:ale_lint_on_text_changed = 'normal'
    let g:ale_linter_aliases = {}
    let g:ale_linters = {'typescript': ['tsserver'], 'json': ['vscodejson'], 'html': ['vscodehtml'], 'rust': ['analyzer'], 'javascript': ['tsserver', 'vscodeeslint'], 'css': ['vscodecss']}
    let g:ale_linters_explicit = 1
    let g:ale_linters_ignore = {}
    let g:ale_list_vertical = 0
    let g:ale_list_window_size = 10
    let g:ale_loclist_msg_format = '%code: %%s'
    let g:ale_max_buffer_history_size = 20
    let g:ale_max_signs = -1
    let g:ale_maximum_file_size = v:null
    let g:ale_open_list = 0
    let g:ale_pattern_options = v:null
    let g:ale_pattern_options_enabled = v:null
    let g:ale_root = {}
    let g:ale_set_balloons = 1
    let g:ale_set_highlights = 1
    let g:ale_set_loclist = 1
    let g:ale_set_quickfix = 0
    let g:ale_set_signs = 1
    let g:ale_sign_column_always = 0
    let g:ale_sign_error = '>>'
    let g:ale_sign_info = '--'
    let g:ale_sign_offset = 1000000
    let g:ale_sign_style_error = '>>'
    let g:ale_sign_style_warning = '--'
    let g:ale_sign_warning = '--'
    let g:ale_sign_highlight_linenrs = 0
    let g:ale_statusline_format = v:null
    let g:ale_type_map = {}
    let g:ale_use_neovim_diagnostics_api = 0
    let g:ale_use_global_executables = v:null
    let g:ale_virtualtext_cursor = 'all'
    let g:ale_warn_about_trailing_blank_lines = 1
    let g:ale_warn_about_trailing_whitespace = 1
     LSP Error Messages:
    
    (Errors for vscodehtml)
    Failed to find project root, language server won't start.
      Command History:

**Comment by w0rp** [comment]
@gnietto Raise an issue for this with what your project looks like. There may be a file in your project we can use to detect the project root automatically.

**Comment by gnietto** [comment, marked as answer]
> I feel I'm missing something very basic here

@w0rp Effectively, that was the case as a new user.
After testing my projects and reading ALE code and documentation, I discovered that vscodehtml threw that message because I did not initialize the project with `git init`.
For me, ALE is a great tool so congratulations to creators and maintainers and thanks for the quick response.
I have other questions about ALE's (auto)completion although I will ask them in another thread of discussions.
